pkg/asset/manifests/powervs/cluster.go

```diff
@@ -0,0 +1,238 @@
+
+	bucketName = fmt.Sprintf("%s-bootstrap-ign", clusterID.InfraID)
+
+	powerVSCluster = &capibm.IBMPowerVSCluster{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: capibm.GroupVersion.String(),
+			Kind:       "IBMPowerVSCluster",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      clusterID.InfraID,
+			Namespace: capiutils.Namespace,
+			Annotations: map[string]string{
+				"powervs.cluster.x-k8s.io/create-infra": "true",
+			},
+		},
+		Spec: capibm.IBMPowerVSClusterSpec{
+			Network: capibm.IBMPowerVSResourceReference{
+				Name: &network,
+			},
+			ServiceInstance: &service,
+			Zone:            &installConfig.Config.Platform.PowerVS.Zone,
+			ResourceGroup: &capibm.IBMPowerVSResourceReference{
+				Name: &resourceGroupName,
+			},
+			VPC: &capibm.VPCResourceReference{
+				Name:   &vpcName,
+				Region: &vpcRegion,
+			},
+			TransitGateway: &capibm.TransitGateway{
+				Name: &transitGatewayName,
+			},
+			LoadBalancers: []capibm.VPCLoadBalancerSpec{
+				{
+					Name:   fmt.Sprintf("%s-loadbalancer", clusterID.InfraID),
+					Public: ptr.To(true),
+					AdditionalListeners: []capibm.AdditionalListenerSpec{
+						{
+							Port: 22,
+						},
+						// @BUG We should be able to specify this:
+						// capibm.AdditionalListenerSpec{
+						//	Port: 6443,
+						// },
+					},
+				},
+				{
+					Name:   fmt.Sprintf("%s-loadbalancer-int", clusterID.InfraID),
+					Public: ptr.To(false),
+					AdditionalListeners: []capibm.AdditionalListenerSpec{
+						// @BUG We should be able to specify this:
+						// capibm.AdditionalListenerSpec{
+						//	Port: 6443,
+						// },
+						{
+							Port: 22623,
+						},
+					},
+				},
+			},
+			CosInstance: &capibm.CosInstance{
+				Name:         cosName,
+				BucketName:   bucketName,
+				BucketRegion: vpcRegion,
+			},
+			Ignition: &capibm.Ignition{
+				Version: "3.4",
+			},
+		},
+	}
+
+	// Avoid:
+	//   vpc:
+	//     name: ""
+	//     region: ""
+	if installConfig.Config.Platform.PowerVS.VPCName != "" {
+		vpcResource := capibm.VPCResourceReference{
+			Name:   &installConfig.Config.Platform.PowerVS.VPCName,
+			Region: &installConfig.Config.Platform.PowerVS.VPCRegion,
+		}
+		reflect.ValueOf(&powerVSCluster.Spec).Elem().FieldByName("VPC").Set(reflect.ValueOf(&vpcResource))
+	}
+
+	// Another avoidence
+	logrus.Debugf("GenerateClusterAssets: len(VPCSubnets) = %d", len(installConfig.Config.Platform.PowerVS.VPCSubnets))
+	if len(installConfig.Config.Platform.PowerVS.VPCSubnets) > 0 {
+		var (
+			subnets   []capibm.Subnet
+			vpcSubnet string
+		)
+
+		subnets = make([]capibm.Subnet, len(installConfig.Config.Platform.PowerVS.VPCSubnets))
+
+		for _, vpcSubnet = range installConfig.Config.Platform.PowerVS.VPCSubnets {
+			subnets = append(subnets, capibm.Subnet{ID: &vpcSubnet})
+		}
+		logrus.Debugf("GenerateClusterAssets: subnets = %+v", subnets)
+
+		reflect.ValueOf(&powerVSCluster.Spec).Elem().FieldByName("VPCSubnets").Set(reflect.ValueOf(subnets))
```

Comment:
```suggestion
		powerVSCluster.Spec.VPCSubnets = make([]capibm.Subnet, len(installConfig.Config.Platform.PowerVS.VPCSubnets))

		for _, vpcSubnet := range installConfig.Config.Platform.PowerVS.VPCSubnets {
			powerVSCluster.Spec.VPCSubnets = append(powerVSCluster.Spec.VPCSubnets, capibm.Subnet{ID: &vpcSubnet})
		}
		logrus.Debugf("GenerateClusterAssets: subnets = %+v", powerVSCluster.Spec.VPCSubnets)
```
I think your code has a bug here, since you're reusing the `vpcSubnet` variable and taking its address for all subnets but the value it points to changes with every iteration. By using `:=` (and Golang 1.21+), we have a different var per iteration.